pybamm/parameters/thermal_parameters.py

```diff
@@ -28,17 +31,40 @@ def __init__(self):
 
     def _set_parameters(self):
         """Defines the dimensional parameters"""
-        for domain in self.domain_params.values():
-            domain._set_parameters()
+        for domain, params in self.domain_params.items():
+            params._domain = domain  # Set _domain attribute
```

Comment:
I don't think this is needed, since `T` is defined across all domains, not by component